### ISSUE TITLE
Stir motor type: stepper or brushed?

### ISSUE THREAD

**Issue by chuck-h** [hivebio/ministat-1#1]
The UW turbidostat uses a stepper motor to turn the magnet hub for stirring. Apparently they recently switched motors and changed the circuit slightly because of supply problems with the first motor.

I suggest we use a small brushed motor like http://www.adafruit.com/products/711 .
- cheaper motor
- simpler drive (one transistor & diode)
- fewer microprocessor pins

Question: do we need speed control on the stirrer, or can we just give it a voltage and let it spin? This motor probably likes to run pretty fast, is that a problem?

**Comment by ptschnack** [comment]
I was thinking about this too. I Googled: 'adafruit speed control dc motor' and got this:
https://learn.adafruit.com/adafruit-motor-selection-guide/dc-motor-control
Upshot: we can use cheap dc motor and drive it with pwm. (Whew! Expensive steppers are overkill for this.)

```
  From: chuck-h <notifications@github.com>
```

 To: hivebio/ministat-1 ministat-1@noreply.github.com 
 Sent: Monday, January 26, 2015 8:35 AM
 Subject: [ministat-1] Stir motor type: stepper or brushed? (#1)

The UW turbidostat uses a stepper motor to turn the magnet hub for stirring. Apparently they recently switched motors and changed the circuit slightly because of supply problems with the first motor.I suggest we use a small brushed motor like http://www.adafruit.com/products/711 .   
- cheaper motor
- simpler drive (one transistor & diode)
- fewer microprocessor pins
  Question: do we need speed control on the stirrer, or can we just give it a voltage and let it spin? This motor probably likes to run pretty fast, is that a problem?—

**Comment by ptschnack** [comment]
The uController has PWM outputs, right? Can these be used to drive a transistor or other power amp to run the motor?

**Comment by chuck-h** [comment]
PWM to a power transistor sounds like a great idea; I think we want about 1V at the motor.

**Comment by zanghell** [comment]
is that to replace a motor controller ? sorry if I missed something in the thread.
Alex.

**Comment by ptschnack** [comment]
Yes. Small DC motor controllers are often just a pulse width modulator driving a power output stage. The uControllers we are using can be used as PWM drivers, so the uController itself can drive the motor. We just need to add a very simple power/buffer stage. Probably just a transistor with protective diode and caps for filtering.

**Comment by chuck-h** [comment]
Last week I received the small stir bars and ran one in water with the DC stir motor. I used a PWM microcontroller output driving one channel of a ULN2003A darlington array. Motor speed and distance from magnets to testube have to be controlled in order to get stable motion, but I think it will work.
